waiter/integration/waiter/kubernetes_scheduler_integration_test.clj

```diff
@@ -337,3 +337,54 @@
                     :interval 15
                     :timeout timeout-secs)))))
         (log/warn "skipping test as INTEGRATION_TEST_BAD_IMAGE is not specified")))))
+
+(deftest ^:parallel ^:integration-fast test-kubernetes-reverse-proxy-sidecar
+  (testing-using-waiter-url
+    (when (using-k8s? waiter-url)
+      (if (contains? (get-kubernetes-scheduler-settings waiter-url) :reverse-proxy)
+        (let [reverse-proxy-flag "REVERSE_PROXY"
+              reverse-proxy-offset 1
+              x-waiter-name (rand-name)
+              _ (log/info "making canary request")
+              {:keys [cookies service-id] :as response} (make-request-with-debug-info
```

Comment:
Nitpick: you can get the `headers` from the response here and re-use them below in your requests instead of having to reconstruct them. Alternatively, you can extract the header-map into `request-headers` variable and use them in all the three places.